### ISSUE TITLE
Assigning CRS to a GeoDataFrame without a geometry...

### ISSUE THREAD

**Issue by msw09090** [noaa-ocs-modeling/adcircpy#155]
Hello,

I used adcircpy a couple weeks ago and I didn't encounter this problem. For whatever reason, recently, I cannot generate tidal constituents given a fort.14 file (+ a duration and start date) using repurposed script (mostly resembles 'example_1.py' from this github page).

The error is generated when 'driver_write' is called: driver.write(outpath, overwrite=True).

The error itself is (seems to be) related to an unspecified CRS: 
`ValueError: Assigning CRS to a GeoDataFrame without a geometry column is not supported. Supply geometry using the 'geometry=' keyword argument, or by providing a DataFrame with column name 'geometry'`

My fort.14 is provided in decimal degrees and I do specify the 'crs=4326.'

I reinstalled adcircpy and I encountered the same problem which seems to be related to geopandas.

I attached (I think?) my script (adcircpy_error.txt), error (adcircpy_error.txt), and fort.14 (fort14.txt) file.
[fort14.txt](https://github.com/noaa-ocs-modeling/adcircpy/files/9768309/fort14.txt)
[adcircpy_error.txt](https://github.com/noaa-ocs-modeling/adcircpy/files/9768316/adcircpy_error.txt)

Thanks in advance!

**Comment by SorooshMani-NOAA** [comment]
@msw09090 this is because you used to have `geopandas` version `<=0.10` before, but recently updated to version `0.11` which says the following in the changelog:
> Specifying a CRS in the GeoDataFrame constructor when no geometry column is provided and calling GeoDataFrame. set_crs on a GeoDataFrame without an active geometry column now raise a ValueError (#2100)

You can find the changelog here:
https://geopandas.org/en/stable/docs/changelog.html

Thank you for bringing this issue to our attention. I suggest that, for now, you explicitly install `geopandas==0.10` and see if you still have any issues, until a proper fix is implemented.

**Comment by SorooshMani-NOAA** [comment]
I'm just pasting a snippet of the traceback here:
```
Input In [7], in <cell line: 1>()
----> 1 driver.write('', overwrite=True)

File ~\Anaconda3\lib\site-packages\adcircpy\driver.py:388, in AdcircRun.write(self, output_directory, overwrite, fort14, fort13, fort22, fort15, coldstart, hotstart, driver, nproc)
    386 self._IHOT = 0
    387 # and call the hotstart writer,
--> 388 super().write('hotstart', output_directory / fort15, overwrite)
    389 if self.wind_forcing is not None:
    390     if fort22:
.
.
.
File ~\Anaconda3\lib\site-packages\adcircpy\mesh\fort14.py:55, in BaseBoundaries.gdf(self)
     44     for i, boundary in enumerate(self._data):
     45         data.append(
     46             {
     47                 'geometry': LineString(
   (...)
     53             }
     54         )
---> 55     self._gdf = gpd.GeoDataFrame(data, crs=self._mesh.crs)
     56 return self._gdf

File ~\Anaconda3\lib\site-packages\geopandas\geodataframe.py:192, in GeoDataFrame.__init__(self, data, geometry, crs, *args, **kwargs)
    189     self.set_geometry(geometry, inplace=True, crs=crs)
    191 if geometry is None and crs:
--> 192     raise ValueError(
    193         "Assigning CRS to a GeoDataFrame without a geometry column is not "
    194         "supported. Supply geometry using the 'geometry=' keyword argument, "
    195         "or by providing a DataFrame with column name 'geometry'",
    196     )

ValueError: Assigning CRS to a GeoDataFrame without a geometry column is not supported. Supply geometry using the 'geometry=' keyword argument, or by providing a DataFrame with column name 'geometry'
```

**Comment by msw09090** [comment]
Hello again,

Thank you for the quick response!

Your 'Quickfix' worked (geopandas version/installation change).

For anyone with the same issue, simply remove your virtual environment then reinstall adcircpy.

**Comment by SorooshMani-NOAA** [comment]
I ran into some issues trying to run automated tests. When they're done I'll merge so that the `main` branch could be used. Later the issue needs to be addressed by updating the instances in the code where empty `GeoDataframe`s are created with `crs` specification. So I'll keep the ticket open for now.

**Comment by msw09090** [comment]
I'm working with a bunch of environments at the moment so I may have accidently upgraded geopandas after initially installing it (adcircpy from earlier today) correctly. I ran adcircpy after my initial test and it failed once again (same error).

Another alternative to fix the problem (for the newbies) is to run:< "python -m pip install geopandas==0.10" >. I did this on a Linux OS.

That worked for me. You'll get a depreciation warning if you do this, but it's a nice patch for now.

**Comment by msw09090** [comment]
I just realized, after doing what I did above, only half of the 'NBFR' fields were generated using the same script/command: tide_gen fort.14 2022-09-26T00:00:00 24 --mesh-crs='epsg:4326' - so I can't run an ADCIRC simulation w/NBFR>0.

I used the same script I ran when I initially created this thread, as well... Same result.

Any idea what this could be? I created a new environment then downgraded geopandas (0.10).

This was the output (part of the fort.15 file):
8                                                               ! NTIF                                - NUMBER OF TIDAL POTENTIAL CONSTITUENTS BEING FORCED starting 2008082300
Q1
0.019256 6.49585E-05 0.695 1.14036 184.236
O1
0.100514 6.75977E-05 0.695 1.14036 84.1505
P1
0.046843 7.25229E-05 0.706 1 85.212
K1
0.141565 7.29212E-05 0.736 1.08673 269.129
N2
0.046398 0.00013788 0.693 0.974027 91.0445
M2
0.242334 0.000140519 0.693 0.974027 350.959
S2
0.112841 0.000145444 0.693 1 0
K2
0.030704 0.000145842 0.693 1.22665 357.691
8
Q1
6.49585E-05 1.14036 184.236
O1
6.75977E-05 1.14036 84.1505
P1
7.25229E-05 1 85.212
K1
7.29212E-05 1.08673 269.129
N2
0.00013788 0.974027 91.0445
M2
0.000140519 0.974027 350.959
S2
0.000145444 1 0
K2
0.000145842 1.22665 357.691
